### ISSUE TITLE
Failure parsing datetime "1-01-01 00:00:00"

### ISSUE THREAD

**Issue by aciduck** [Azure/azure-kusto-python#154]
Running the ".show diagnostics" command sometimes returns a value of "1-01-01 00:00:00" for the DataWarmingLastRunOn column. This value throws an exception when it is being converted to a python object:
> Traceback (most recent call last):
  File "/XXX/kusto_monitor.py", line 192, in <module>
    main()
  File "/XXX/kusto_monitor.py", line 186, in main
    process_cluster(cluster, args, output_file)
  File "/XXX/kusto_monitor.py", line 141, in process_cluster
    log_command_results(client, cluster_name, NO_DATABASE, ".show diagnostics", "diagnostics", output_file)
  File "/XXX/kusto_monitor.py", line 93, in log_command_results
    query_result = client.execute(database_name, command)
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/request.py", line 395, in execute
    return self.execute_mgmt(database, query, properties)
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/request.py", line 418, in execute_mgmt
    return self._execute(self._mgmt_endpoint, database, query, None, KustoClient._mgmt_default_timeout, properties)
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/request.py", line 463, in _execute
    return KustoResponseDataSetV1(response.json())
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/_response.py", line 110, in __init__
    super(KustoResponseDataSetV1, self).__init__(json_response["Tables"])
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/_response.py", line 18, in __init__
    self.tables = [KustoResultTable(t) for t in json_response]
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/_models.py", line 137, in __init__
    self.rows = [KustoResultRow(self.columns, row) for row in json_table["Rows"]]
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/_models.py", line 67, in __init__
    self._hidden_values.append(to_pandas_datetime(value))
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/azure/kusto/data/helpers.py", line 8, in to_pandas_datetime
    return pd.to_datetime(raw_value)
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/pandas/core/tools/datetimes.py", line 469, in to_datetime
    result = _convert_listlike(np.array([arg]), box, format)[0]
  File "/XXX/venv_healthcheck/local/lib/python2.7/site-packages/pandas/core/tools/datetimes.py", line 380, in _convert_listlike
    raise e
pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 1-01-01 00:00:00

**Comment by danield137** [comment]
Hi @aciduck,  Thanks for reporting!
While I check this internally, could you kindly provide version numbers (`pip freeze` output)?

**Comment by aciduck** [comment]
Here are all the packages:

adal==1.2.0
APScheduler==2.1.2
asn1crypto==0.24.0
azure==4.0.0
azure-applicationinsights==0.1.0
azure-batch==4.1.3
azure-common==1.1.16
azure-cosmosdb-nspkg==2.0.2
azure-cosmosdb-table==1.0.5
azure-datalake-store==0.0.39
azure-eventgrid==1.2.0
azure-graphrbac==0.40.0
azure-keyvault==1.1.0
azure-kusto-data==0.0.31
azure-loganalytics==0.1.0
azure-mgmt==4.0.0
azure-mgmt-advisor==1.0.1
azure-mgmt-applicationinsights==0.1.1
azure-mgmt-authorization==0.50.0
azure-mgmt-batch==5.0.1
azure-mgmt-batchai==2.0.0
azure-mgmt-billing==0.2.0
azure-mgmt-cdn==3.0.0
azure-mgmt-cognitiveservices==3.0.0
azure-mgmt-commerce==1.0.1
azure-mgmt-compute==4.3.1
azure-mgmt-consumption==2.0.0
azure-mgmt-containerinstance==1.4.0
azure-mgmt-containerregistry==2.4.0
azure-mgmt-containerservice==4.2.2
azure-mgmt-cosmosdb==0.4.1
azure-mgmt-datafactory==0.6.0
azure-mgmt-datalake-analytics==0.6.0
azure-mgmt-datalake-nspkg==3.0.1
azure-mgmt-datalake-store==0.5.0
azure-mgmt-datamigration==1.0.0
azure-mgmt-devspaces==0.1.0
azure-mgmt-devtestlabs==2.2.0
azure-mgmt-dns==2.1.0
azure-mgmt-eventgrid==1.0.0
azure-mgmt-eventhub==2.2.0
azure-mgmt-hanaonazure==0.1.1
azure-mgmt-iotcentral==0.1.0
azure-mgmt-iothub==0.5.0
azure-mgmt-iotcentral==0.1.0                                                                                                                                        [46/107]
azure-mgmt-iothub==0.5.0
azure-mgmt-iothubprovisioningservices==0.2.0
azure-mgmt-keyvault==1.1.0
azure-mgmt-loganalytics==0.2.0
azure-mgmt-logic==3.0.0
azure-mgmt-machinelearningcompute==0.4.1
azure-mgmt-managementgroups==0.1.0
azure-mgmt-managementpartner==0.1.0
azure-mgmt-maps==0.1.0
azure-mgmt-marketplaceordering==0.1.0
azure-mgmt-media==1.0.0
azure-mgmt-monitor==0.5.2
azure-mgmt-msi==0.2.0
azure-mgmt-network==2.4.0
azure-mgmt-notificationhubs==2.0.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-policyinsights==0.1.0
azure-mgmt-powerbiembedded==2.0.0
azure-mgmt-rdbms==1.5.0
azure-mgmt-recoveryservices==0.3.0
azure-mgmt-recoveryservicesbackup==0.3.0
azure-mgmt-redis==5.0.0
azure-mgmt-relay==0.1.0
azure-mgmt-reservations==0.2.1
azure-mgmt-resource==2.0.0
azure-mgmt-scheduler==2.0.0
azure-mgmt-search==2.0.0
azure-mgmt-servicebus==0.5.3
azure-mgmt-servicefabric==0.2.0
azure-mgmt-signalr==0.1.1
azure-mgmt-sql==0.9.1
azure-mgmt-storage==2.0.0
azure-mgmt-subscription==0.2.0
azure-mgmt-trafficmanager==0.50.0
azure-mgmt-web==0.35.0
azure-nspkg==3.0.2
azure-servicebus==0.21.1
azure-servicefabric==6.3.0.0
azure-servicemanagement-legacy==0.20.6
azure-storage-blob==1.4.0
azure-storage-common==1.4.0
azure-storage-file==1.4.0
azure-storage-nspkg==3.1.0
azure-storage-queue==1.4.0
boto==2.49.0
certifi==2018.11.29
cffi==1.11.5
certifi==2018.11.29                                                                                                                                                  [0/107]
cffi==1.11.5
chardet==3.0.4
Click==7.0
cryptography==2.2.2
decorator==4.3.0
docopt==0.4.0
elasticsearch==2.0.0
elasticsearch-curator==3.4.0
enum34==1.1.6
future==0.17.1
futures==3.2.0
httplib2==0.12.0
idna==2.7
ipaddress==1.0.22
isodate==0.6.0
mandrill==1.0.57
msrest==0.6.2
msrestazure==0.6.1
ndg-httpsclient==0.5.1
npm==0.1.1
numpy==1.15.4
O365==0.9.16
oauthlib==2.1.0
optional-django==0.1.0
pandas==0.23.4
pathlib2==2.3.3
pika==0.9.14
py==1.7.0
py-zabbix==1.1.3
pyasn1==0.4.4
PyChef==0.3.0
pycparser==2.19
pycurl==7.43.0.2
PyJWT==1.7.1
pymongo==3.6.1
pyOpenSSL==18.0.0
python-dateutil==2.8.0
pytz==2018.7
requests==2.22.0
requests-oauthlib==1.0.0
retry==0.9.2
scandir==1.9.0
six==1.11.0
typing==3.6.6
urllib3==1.24.1
ws4py==0.5.1

**Comment by danield137** [comment]
Closing, couldn't repro. re-open if still relevant.

**Comment by castle8080** [comment]
I am not sure what the best solution for this problem is. I would imagine that changing the type used for datetimes would break many people's code. For now I made my own version of dataframe_from_result_table in my code base where I added the option errors='coerce' where pd.to_datetime is called. Instead of getting an exception, the out of bounds values are converted to NaT (not a time). Exposing some error handling options on dataframe_from_result_table would be easy and helpful.
